addons/test_mail/tests/test_mail_composer.py

```diff
@@ -659,6 +659,29 @@ def test_mail_composer_wtpl_complete(self):
         self.assertEqual(set(message.attachment_ids.mapped('res_id')), set(self.test_record.ids))
         self.assertTrue(all(attach not in message.attachment_ids for attach in attachs), 'Should have copied attachments')
 
+    def test_mail_composer_document_based(self):
```

Comment:
Could you move test definition up ? As tests are executed alphabetically in a given class, better order them alphabetically so that test execution order matches definition order. That way when you have several issues, you can follow console errors in the file ^^ .